src/application-manager.coffee

```diff
@@ -142,11 +143,25 @@ module.exports = class ApplicationManager extends EventEmitter
 			cleanup: (step) =>
 				@images.cleanup()
 			createNetworkOrVolume: (step) =>
-				model = if step.model is 'volume' then @volumes else @networks
-				model.create(step.target)
+				if step.model is 'network'
+					# TODO: These step targets should be the actual compose objects,
+					# rather than recreating them
+					Network.fromComposeObject({ @docker, @logger },
```

Comment:
I see you have a lot of Network.fromComposeObject calls in several places. Wouldn't it be better to just call this when you're getting the current and target state (i.e. in getCurrentForComparison, and in getTarget), and then always work with Network objects? This is what we do with Service (we create Service objects when we get the current and target states, and therefore only call `new Service` or `Service.fromWhatever` in those two specific places.

Reply:
Good point, I should have made that a FIXME, not a TODO because I did plan to tacklet it - thanks dude!